### PR TITLE
fix: Update dependencies to avoid build failure

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What changes did you make? (Give an overview)

Upgraded the following packages to allow `npm install` to work without error:

- typescript
- rollup-plugin-dts
- rollup

#### Related Issues

refs #630

#### Is there anything you'd like reviewers to focus on?

@brettz9 can you help me fix the failing test? I can't for the life of me figure out what it's trying to do, but it appears to not be handling `TSArrayType`s properly.

### CODE REVIEW

**Comment by snitin315** [comment]
>Upgraded the following packages to allow npm install to work without error:

I see only ESLint updated in package.json.

**Comment by nzakas** [comment]
Oops, looks like I missed a commit.

**Comment by nzakas** [comment]
Now this is super weird, everything locally is working but the CI isn't. ☹️

**Comment by brettz9** [comment]
Yeah, I'm getting install errors before being able to look at tests.

**Comment by brettz9** [comment]
When I amend the `tsc` script to add `tsc -v`, it gives "Version 4.5.5", despite `package.json` having the ^5.6.2 version. That's where the warning is apparently coming from. I guess you don't want `build:types` as part of the `prepare` routine for the sake of first-time installs.

**Comment by nzakas** [comment]
Yeah, I've tried removing `prepare` altogether but I'm still getting the same error. I'll push that commit.

**Comment by brettz9** [comment]
Even weirder is that when I remove that from the install script, the install proceeds and completes, saying that typescript 5.6.2 is installed but typescript is not present in `node_modules`.

**Comment by nzakas** [comment]
I'm also seeing that the reported `tsc` version locally is 4.5.5, yet when I go into `node_modules` and look at the `typescript` package, it clearly says 5.6.2. 🤯 

I can find any evidence of any package installing 4.5.5, though.

**Comment by brettz9** [comment]
The different set of packages in `node_modules` is changed to include the proper typescript version when I remove:

```json
"workspaces": [
    "packages/*"
  ],
```

...though I know that's the whole point.

**Comment by nzakas** [comment]
Okay, I think I know what's happening. The packages are installed in alphabetical order, so something in `eslint-scope` is causing the old TypeScript to be installed. Commit incoming.

**Comment by nzakas** [comment]
Well, different error, but old `tsc` is still being installed. 🤔

**Comment by brettz9** [comment]
Could this be of significance?

After running `npm i -ws`, I get:

```
npm warn workspaces eslint-scope in filter set, but no workspace folder present
npm warn workspaces eslint-visitor-keys in filter set, but no workspace folder present
npm warn workspaces espree in filter set, but no workspace folder present
```

**Comment by brettz9** [comment]
`tsd` is installing the bad TS version (one reason I like having package lock files built to see this kind of info easily)

**Comment by brettz9** [comment]
Oddly, when upgrading tsd, I get an even older version of typescript showing up within package.json `tsc -v`; it is apparently using my global TS version now.

**Comment by brettz9** [comment]
The TS version is fixed when I change this in the root package.json:

```json
  "workspaces": [
    "packages/eslint-scope",
    "packages/eslint-visitor-keys",
    "packages/eslint-espree"
  ],
```

**Comment by nzakas** [comment]
The top-level `package.json` is working as-is on CI. Just need to tweak the test script to include `build` since I removed `prepare`.

So I think *now* I can ask you about the failing test in `eslint-visitor-keys`. :smile:

**Comment by brettz9** [comment]
Ok, I can look into it, but it may be working on CI merely because it has a higher global TS version installed. With the change I made, the warnings like this are now gone also:

> npm warn workspaces eslint-visitor-keys in filter set, but no workspace folder present

**Comment by brettz9** [comment]
I'm afraid this was a bit fragile to begin with, and now with updates, the script can't accommodate. I think you'll either need to build the keys again manually or find someone with greater bandwidth than myself to adapt the script. I'm finding it difficult to revisit the code unfortunately.

**Comment by nzakas** [comment]
@brettz9 okay, thanks.

**Comment by nzakas** [comment]
I've gone through and remove the tools because they don't work anymore. I think we're okay to update the keys manually when needed.

File: packages/eslint-visitor-keys/package.json [hunk 1]
```diff
@@ -28,26 +28,24 @@
...
+    "tsd": "^0.31.2",
+    "typescript": "^5.6.2"
   },
   "scripts": {
     "build": "npm run build:cjs && npm run build:types",
     "build:cjs": "rollup -c",
     "build:debug": "npm run build:cjs -- -m && npm run build:types",
-    "build:keys": "node tools/build-keys-from-ts",
-    "build:types": "tsc",
-    "prepare": "npm run build",
```
Comment (mdjermanovic) [inline packages/eslint-visitor-keys/package.json:50]: If we just remove the `prepare` script, I believe `npm run build` won't be run during releases.

Comment (nzakas) [reply packages/eslint-visitor-keys/package.json:50]: We already run `build` during releases:
https://github.com/eslint/js/blob/dd7fcbbbe72b834999c9b490b8278cc1aa7209d7/.github/workflows/release-please.yml#L39-L41

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**